packages/gensx/tests/array.test.tsx

```diff
@@ -0,0 +1,115 @@
+import { expect, test } from "vitest";
+
+import { gsx } from "../src";
+
+interface NumberWrapperOutput {
+  value: number;
+}
+
+const NumberWrapper = gsx.Component<{ n: number }, NumberWrapperOutput>(
+  "NumberWrapper",
+  ({ n }) => ({ value: n }),
+);
+
+const NumberDoubler = gsx.Component<{ value: number }, number>(
+  "NumberDoubler",
+  ({ value }) => value * 2,
+);
+
+const AsyncNumberFilter = gsx.Component<{ value: number }, boolean>(
+  "AsyncNumberFilter",
+  async ({ value }) => {
+    await new Promise(resolve => setTimeout(resolve, 1));
+    return value > 5;
+  },
+);
+
+const Value = gsx.Component<{ value: number }, number>(
+  "Value",
+  ({ value }) => value,
+);
+
+const TaskGenerator = gsx.Component<{ value: number }, number[]>(
+  "TaskGenerator",
+  async ({ value }) => {
+    // Simulate a workflow step that generates multiple tasks
+    await new Promise(resolve => setTimeout(resolve, 1));
+    // For each input, generate [value, value*2] as sub-tasks
+    return [value, value * 2];
+  },
+);
+
```

Comment:
I think we need tests for a component that uses `execute` inside of it, and a component that returns another component.

Reply:
👍 updated the taskGenerator as suggested below to cover case (2) and will add a test for (1) as well.

Reply:
Commit https://github.com/cortexclick/gensx/pull/153/commits/bef9c9d9ca5e4390c2721a2db81d5be4dc5caf20 adds tests for: 

1. Components that return components
2. Using gsx.array within a component
3. Using gsx.array within a child function.